### PR TITLE
Fix UMA resource restrictions

### PR DESCRIPTION
**Pull request by ashensw**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 1]
```diff
@@ -354,4 +354,29 @@ private URI getResourceLocationURI(CreateResourceDTO response) throws URISyntaxE
...
+        try {
+            if (resourceOwner != null && consumerKey != null) {
+                List<String> resourceList = ResourceUtils.getResourceService()
+                        .getResourceIds(resourceOwner, userDomain, consumerKey);
+                if (resourceList.contains(resourceId)) {
+                    return true;
+                }
+            }
+        } catch (UMAException e) {
+            handleErrorResponse("Error when retrieving resource for client: " + consumerKey, e);
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: is this line meaning full ?

[Error] Error when retrieving resource for client : null
or
[Error] Error when retrieving resource for client : 34820hnbfsamf023u41qwe

File: components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 2]
```diff
@@ -354,4 +354,29 @@ private URI getResourceLocationURI(CreateResourceDTO response) throws URISyntaxE
...
+        String resourceOwner = null;
+        if (userNameWithDomain != null) {
+            resourceOwner = UserCoreUtil.removeDomainFromName(userNameWithDomain);
+            userDomain = IdentityUtil.extractDomainFromName(userNameWithDomain);
+        }
+
+        try {
+            if (resourceOwner != null && consumerKey != null) {
+                List<String> resourceList = ResourceUtils.getResourceService()
+                        .getResourceIds(resourceOwner, userDomain, consumerKey);
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Looks this is inefficient, getting all resources and then check if one exists.
Need to improve performance.

File: components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java (hunk 1 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: PR updated with the fix.

File: components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java (hunk 2 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: PR updated with the fix. Used a direct DB call to get the required detail.

File: components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/ResourceService.java [hunk 3]
```diff
@@ -41,4 +41,7 @@ Resource getResourceById(String resourceId)
 
     boolean updateResource(String resourceId, Resource resourceRegistration)
             throws UMAServerException, UMAClientException;
+
+    boolean isResourceBelongsToUser(String resourceId, String userName, String userDomain, String consumerKey)
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/ResourceService.java]: would not following method name better?
isResourceOwner(...)

File: components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/dao/ResourceDAO.java [hunk 4]
```diff
@@ -385,4 +385,37 @@ private static Integer checkResourceExistence(String resourceId) throws DataAcce
...
+    /**
+     * Check whether the resource belongs to the given userName
+     *
+     * @param resourceId resource ID of the resource.
+     * @param userName   name of the user.
+     * @param userDomain user store domain of the user.
+     * @param clientId   client id representing the resource server.
+     * @throws UMAServerException
+     */
+
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/dao/ResourceDAO.java]: unnecessary line

Comment (ashensw) [reply components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/dao/ResourceDAO.java]: Fixed

File: components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/ResourceService.java (hunk 3 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/ResourceService.java]: Changed the name and updated the PR.

**Pull request merged**